sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client_helpers.py

```diff
@@ -1000,14 +1003,16 @@ def _commit_block_list_options(
     """
     block_lookup = BlockLookupList(committed=[], uncommitted=[], latest=[])
     for block in block_list:
-        if isinstance(block, BlobBlock):
+        if isinstance(block, BlobBlock) and isinstance(block.state, Enum):
```

Comment:
This seems like an issue with the typing on our `BlobBlock` model? `BlobBlock.state` is of type `BlockState` which is an Enum so it shouldn't need this `isinstance`. Can you check what's up with the model?

Reply:
Fixed core issue at the model-level